DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Receive/RhinoHostObjectBuilder.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Rhino;
+using Rhino.DocObjects;
+using Rhino.Geometry;
+using Speckle.Connectors.Utils.Builders;
+using Speckle.Converters.Common;
+using Speckle.Core.Models;
+using Speckle.Core.Models.Extensions;
+
+namespace Speckle.Connectors.Rhino7.Operations.Receive;
+
+public class RhinoHostObjectBuilder : IHostObjectBuilder
+{
+  private readonly ISpeckleConverterToHost _toHostConverter;
+  private readonly IConversionContextStack<RhinoDoc, UnitSystem> _contextStack;
+
+  public RhinoHostObjectBuilder(
+    ISpeckleConverterToHost toHostConverter,
+    IConversionContextStack<RhinoDoc, UnitSystem> contextStack
+  )
+  {
+    _toHostConverter = toHostConverter;
+    _contextStack = contextStack;
+  }
+
+  public IEnumerable<string> Build(
+    Base rootObject,
+    string projectName,
+    string modelName,
+    Action<string, double?>? onOperationProgressed,
+    CancellationToken cancellationToken
+  )
+  {
+    var baseLayerName = $"Project {projectName}: Model {modelName}";
+
+    var objectsToConvert = rootObject.TraverseWithPath(obj => obj is not Collection);
+
+    var convertedIds = BakeObjects(objectsToConvert, baseLayerName, onOperationProgressed, cancellationToken);
+
+    _contextStack.Current.Document.Views.Redraw();
+
+    return convertedIds;
+  }
+
+  private List<string> BakeObjects(
+    IEnumerable<(List<string>, Base)> objects,
+    string baseLayerName,
+    Action<string, double?>? onOperationProgressed,
+    CancellationToken cancellationToken
+  )
+  {
+    // LET'S FUCK AROUND AND FIND OUT
+    var rootLayerName = baseLayerName;
+    // POC: This Find method was flagged as obsolete and I found no obvious way to work around it.
+    // Silencing it for now but we should find a way to fix this.
+#pragma warning disable CS0618 // Type or member is obsolete
+    var rootLayerIndex = _contextStack.Current.Document.Layers.Find(rootLayerName, true);
+#pragma warning restore CS0618 // Type or member is obsolete
+
+    if (rootLayerIndex >= 0)
+    {
+      foreach (var layer in _contextStack.Current.Document.Layers[rootLayerIndex].GetChildren())
+      {
+        _contextStack.Current.Document.Layers.Purge(layer.Index, false);
```

Comment:
What are we doing here, removing everything from he layers specified by rootLayerIndex?

Reply:
There was this comment previously in there which is the perfect answer to your q: `// Cleans up any previously received objects`